api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this reference.
+	string name = 1;
```

Comment:
@aaronlehmann ultimately this comes from the fact that the way we want people to interact with secrets is using a `name` and not an ID. Everything else comes from that (using `Name` in `SecretReference` so a human can man to the actual secret on `secrets ls`).